code/Model/SiteTree.php

```diff
@@ -688,15 +688,17 @@ public function RelativeLink($action = null)
             $base = $this->URLSegment;
         }
 
-        $this->extend('updateRelativeLink', $base, $action);
-
         // Legacy support: If $action === true, retain URLSegment for homepages,
         // but don't append any action
         if ($action === true) {
             $action = null;
         }
 
-        return Controller::join_links($base, '/', $action);
+        $link = Controller::join_links($base, '/', $action);
+
+        $this->extend('updateRelativeLink', $link, $base, $action);
```

Comment:
Add link as the last argument. Sure this is targeting a major release but we shouldn't make breaking changes for the hell of it.

Reply:
Actually, the correct hook should only have the $link parameter, because $base and $action are irrelevant at that point, right?

Reply:
They both provide context that might be used in the logic of the extension hook.
They're also both already there so any hook that currently uses them will expect them there.

Reply:
Hm. True. But sny existing extension that is using this would expect for lunk and action to be concatenated afterwards. So it's a breaking change anyway.

Reply:
I have moved $link to last

Reply:
> any existing extension that is using this would expect for link and action to be concatenated afterwards

That's a good point. Maybe it would be better to have `$link` first then. Though I'm not overly fussed either way. If you want to put it at the beginning again then feel free to do so - otherwise just comment that you will leave it. Once you've reacted either way I'll take a quick look to see if any core logic uses this hook (I doubt it but worth checking) and then merge.

We'll need some words about this in the changelog but the PR that adds the 5.0.0 changelog file hasn't been merged yet. I won't let that hold up this change - instead I'll make a note for myself to add that to the changelog once it exists.

Reply:
Thanks. I have switched them back.